### PR TITLE
[aptos-tool] Move tool to using Clapv3 instead of Structopt, and add ExtractPeer tool

### PR DESCRIPTION
**Pull request by gregnazario**
This should get us the latest and greatest and allow us to integrate with the latest move changes.  I added ExtractPeer to ensure I knew how to add new tools with clap.

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]

File: crates/aptos/src/op/mod.rs [hunk 1]
```diff
@@ -7,21 +7,22 @@
...
 use crate::CliResult;
-use structopt::StructOpt;
+use clap::{ArgEnum, Subcommand};
 
 pub mod key;
 
 /// CLI tool for performing operational tasks
 ///
-#[derive(Debug, StructOpt)]
+#[derive(Debug, ArgEnum, Subcommand)]
```
Comment (davidiw) [inline crates/aptos/src/op/mod.rs:16]: alpha ordering 😭

File: crates/aptos/src/op/key.rs [hunk 2]
```diff
@@ -147,14 +179,20 @@ impl SaveKey {
...
 
         // Write private and public keys to files
-        write_to_file(&self.key_file, key_name, encoded_private_key)?;
-        write_to_file(&self.public_key_file()?, key_name, encoded_public_key)
+        let public_key_file = self.public_key_file()?;
+        write_to_file(&self.key_file, key_name, &encoded_private_key)?;
+        write_to_file(&public_key_file, key_name, &encoded_public_key)?;
+
+        let mut map = HashMap::new();
+        map.insert("PrivateKey Path", self.key_file.clone());
```
Comment (davidiw) [inline crates/aptos/src/op/key.rs:193]: consts maybe?

File: crates/aptos/src/op/key.rs [hunk 3]
```diff
@@ -173,11 +211,11 @@ pub fn encode_key<Key: ValidCryptoMaterial>(
...
-/// Write a `Vec<u8>` to a file
-fn write_to_file(key_file: &Path, key_name: &str, encoded_key: Vec<u8>) -> Result<(), Error> {
-    let mut file = File::create(key_file).map_err(|e| Error::IO(key_name.to_string(), e))?;
-    file.write_all(&encoded_key)
-        .map_err(|e| Error::IO(key_name.to_string(), e))
+/// Write a `&[u8]` to a file
+fn write_to_file(key_file: &Path, name: &str, encoded: &[u8]) -> Result<(), Error> {
+    let mut file = File::create(key_file).map_err(|e| Error::IO(name.to_string(), e))?;
+    file.write_all(&encoded)
+        .map_err(|e| Error::IO(name.to_string(), e))
```
Comment (davidiw) [inline crates/aptos/src/op/key.rs:218]: consider std.fs.write

**Pull request merged**